pkg/compare/parsing.go

```diff
@@ -309,3 +318,36 @@ func parseDiffConfig(filePath string) (UserConfig, error) {
 	err = parseYaml(os.DirFS("/"), confPath[1:], &result, userConfNotExistsError, userConfigNotInFormat)
 	return result, err
 }
+
+func (u *UserConfig) getBestMatchFunc() func(templates []*ReferenceTemplate, CR *unstructured.Unstructured, o *Options) (*ReferenceTemplate, *bytes.Buffer, error) {
+	if u.BestMatchType == None {
+		return getOnlyIfOneMatch
+	}
+	return getBestMatchByLines
+}
+
+func getBestMatchByLines(templates []*ReferenceTemplate, cr *unstructured.Unstructured, o *Options) (*ReferenceTemplate, *bytes.Buffer, error) {
+	var bestTemp *ReferenceTemplate
+	minDiffNum := math.MaxInt
+	var minDiffOutput *bytes.Buffer
+	for _, temp := range templates {
+		diffOutput, err := diffAgainstTemplate(temp, cr, o)
+		if err != nil {
+			return nil, minDiffOutput, err
+		}
+		minDiffNum = min(bytes.Count(diffOutput.Bytes(), []byte("\n")), minDiffNum)
```

Comment:
Wondering if `+` and `-` at the beginning of a line would be a better measure? If diffs dispersed throughout the manifest instead of in one clump the could throw off the count as the context the diff tool gives will inflate the number of newlines.

Reply:
Perhaps using something like generating a merge patch and counting the number of leaf nodes in the yaml would be more tool agnostic metric.